### ISSUE TITLE
Allow comments in the `local.settings.json` file

### ISSUE THREAD

**Issue by rafaelldi** [JetBrains/azure-tools-for-intellij#900]
Here https://github.com/JetBrains/azure-tools-for-intellij/blob/develop/PluginsAndFeatures/azure-toolkit-for-rider/azure-intellij-plugin-appservice-dotnet/src/main/kotlin/com/microsoft/azure/toolkit/intellij/legacy/function/localsettings/FunctionLocalSettingsService.kt#L30
we should add `allowComments` https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-builder/allow-comments.html

**Comment by RubenDelange** [comment]
Hi @rafaelldi, do you have any ETA on this one?

**Comment by rafaelldi** [comment]
Unfortunately not, because here we are dependent on IntelliJ Platform. When the `kotlinx-serialization-json` package will be updated in the Platform, we will be able to use this property. This might happen in 2024.3, but I can't guarantee it.

**Comment by nathan-plotts** [comment]
@RubenDelange if you're working in .net and all your comments are on their own line then I was able to create a workaround build task that removes those lines from the local.settings.json file in your bin directory. You'll need to add the 2 sections in red to your project file:
![buildtask](https://github.com/user-attachments/assets/cf8be9e6-8f81-4dc5-95a6-7b5f782af34a)

```
  <UsingTask TaskName="RemoveJsonComments" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
    <ParameterGroup>
      <FilePath ParameterType="System.String" Required="true"/>
    </ParameterGroup>
    <Task>
      <Using Namespace="System.IO"/>
      <Using Namespace="System.Linq"/>
      <Code Type="Fragment" Language="cs"><![CDATA[
var lines = File.ReadAllLines(FilePath);
var linesWithoutComments = lines.Where(l => !l.TrimStart().StartsWith("//"));
File.WriteAllLines(FilePath, linesWithoutComments);

]]></Code>
    </Task>
  </UsingTask>
```

```
  <Target Name="RemoveCommentsFromConfig" AfterTargets="Build">
    <Message Importance="high" Text="Started removing comments from config file $(TargetDir)local.settings.json" />
    <RemoveJsonComments FilePath="$(TargetDir)local.settings.json" />
    <Message Importance="high" Text="Completed removing comments from config file $(TargetDir)local.settings.json" />
  </Target>
```

**Comment by wislon** [comment]
@rafaelldi me again haha, I've noticed this as well, suddenly, since upgrading to the Azure Toolkit for Rider v4.2.1 (where you helpfully fixed the 'can't find certificate' error). 

It didn't used to do this before though, I've been launching functions with comments in the `local.settings.json` for at least the last 18 months, even before. would a downgrade of Rider to something earlier than 2024.2.4 help with this? Or is there perhaps a workaround?

Edit to update: downgrading back as far as 2024.2.3 didn't resolve the issue, also tried downgrading the Azure Toolkit for Rider plugin but then started running into the 'invalid certificate' error again, at which point I gave up.

I've created an issue for Rider as well, maybe that might help get it some traction: https://youtrack.jetbrains.com/issue/RIDER-118306/Unable-to-Launch-Azure-Functions-Locally-With-Comments-in-local.settings.json
